q2_demux/_summarize/_visualizer.py

```diff
@@ -55,50 +55,47 @@ def _link_sample_n_to_file(files, counts, subsample_ns):
 
 def _subsample_paired(fastq_map):
     qual_sample = collections.defaultdict(list)
-    seq_len = None
+    min_seq_len = {}
+    min_seq_len['forward'] = float('inf')
+    min_seq_len['reverse'] = float('inf')
+    max_seq_len = 0
     for fwd, rev, index in fastq_map:
         file_pair = zip(_read_fastq_seqs(fwd), _read_fastq_seqs(rev))
         for i, (fseq, rseq) in enumerate(file_pair):
-            if seq_len is None:
-                seq_len = len(fseq[1])
-            if seq_len != len(fseq[1]):
-                raise ValueError(inconsistent_length_template
-                                 % (seq_len, len(fseq[1])))
-            if seq_len != len(rseq[1]):
-                raise ValueError(inconsistent_length_template
-                                 % (seq_len, len(rseq[1])))
+            min_seq_len['forward'] = min(min_seq_len['forward'], len(fseq[1]))
+            min_seq_len['reverse'] = min(min_seq_len['reverse'], len(rseq[1]))
+            max_seq_len = min(max_seq_len, len(fseq[1]), len(rseq[1]))
             if i == index[0]:
                 qual_sample['forward'].append(_decode_qual_to_phred33(fseq[3]))
                 qual_sample['reverse'].append(_decode_qual_to_phred33(rseq[3]))
                 index.pop(0)
                 if len(index) == 0:
                     break
-
-    return qual_sample
+    return qual_sample, min_seq_len
 
 
 def _subsample_single(fastq_map):
     qual_sample = collections.defaultdict(list)
-    seq_len = None
+    min_seq_len = {}
+    min_seq_len['forward'] = float('inf')
+    min_seq_len['reverse'] = 'null'
```

Comment:
Let's use `None` here, instead --- this will serialize appropriately with the changes requested below in this review.